src/charm.py

```diff
@@ -368,6 +411,7 @@ def _on_upgrade_charm(self, _event: UpgradeCharmEvent):
             _event: required by ops framework, not used.
         """
         self._setup_replica_data(_event)
+        self._update_database()
```

Comment:
What if this fails?

Reply:
Is logging with ERROR  a good idea for this case?

Reply:
Not sure if this is a reason to go into error state. I think it is a bit tricky...

Reply:
Update db fail would be catastrophic, error would auto-retry which is not what we want... @jdkandersson what do we want to do in such cases?

Reply:
David said we should not put this in the `on-upgrade` event. I have removed that in the latest commit

Reply:
As discussed in the APAC Stand-up, you should probably remove this as it's out of scope for this task.